### ISSUE TITLE
Exported metric and stats file do not have same values for number of connected clients

### ISSUE THREAD

**Issue by TopeltR** [Dev25/mcrouter_exporter#20]
Hello! 

We have been using this exporter in our production environment for quite some time, but lately have noticed that the number of connected clients do not match.

```
root@mcrouter:~# curl http://localhost:9108/metrics | grep clients 
mcrouter_clients 0
```

vs 

```
root@mcrouter:~# cat /var/mcrouter/stats/libmcrouter.mcrouter.11211.stats | grep client
"libmcrouter.mcrouter.11211.num_client_connections": 2,
```

When I took a look at the command args metric I also spotted the stats-root being passed, but inside the code, I do not see any parsing or reading done from this. Maybe somebody has an idea why the numbers are constantly different?

**Comment by apolloFER** [comment]
I'm seeing the same issue. I think the reason is: https://github.com/Dev25/mcrouter_exporter/blob/1cd166f0d411963ee3f91b24e824ff10145e64ac/main.go#L532
Based on yours (and mine) stats output, the correct variable to parse would be `num_client_connections`. That's for mcrouter 0.40.0. We also have an older mcrouter cluster running (0.20.0) and back then the stats variable was `libmcrouter.mcrouter.11211.num_clients`

**Comment by apolloFER** [comment]
It appears it changed in 0.39.0 - https://github.com/facebook/mcrouter/commit/c1d34c98b4c2c6f8558364bb990bd07fb3770e09

**Comment by TopeltR** [comment]
This seems to be the case indeed, good catch! I can open up a PR, but how would such changes take place. Should there be some feature flag to change the variable or the exporter master is just in the line with mcrouter latest version?

**Comment by apolloFER** [comment]
I think the correct way would be to parse `e.Version` to SemVer value and compare if it's >= or < `0.39.0`. Also, `mcrouter_clients` is reported as `counter` while it should be a `gauge` - mcrouter increases/decreases that value when clients connect/disconnect.

**Comment by Dev25** [comment]
Good catch, would be happy to review a PR.

The above suggestion would be ideal (check `e.Version` and return old/new key depending on version), however if recent versions of mcrouter have dramatically changed stats i would be happy to cut a breaking release that is only for newer mcrouter versions.

**Comment by Dev25** [comment]
Closed by #26
